odbc_fdw.c

```diff
@@ -598,60 +837,35 @@ static void odbcConnStr(StringInfoData *conn_str,
  * get table size of a table
  */
 static void
-odbcGetTableSize(char *svr_dsn, char *svr_driver, char *svr_host, char *svr_port,
-                 char *svr_database, char *svr_schema, char *svr_table,
-                 char *username, char *password, char *sql_count, unsigned int *size)
+odbcGetTableSize(odbcFdwOptions* options, unsigned int *size)
 {
 	SQLHENV env;
 	SQLHDBC dbc;
 	SQLHSTMT stmt;
 	SQLRETURN ret;
 
-	StringInfoData  conn_str;
 	StringInfoData  sql_str;
-	SQLCHAR OutConnStr[1024];
-	SQLSMALLINT OutConnStrLen;
 
 	SQLUBIGINT table_size;
 	SQLLEN indicator;
 
 	StringInfoData name_qualifier_char;
 	StringInfoData quote_char;
 
-	const char* schema_name = svr_schema;
+	const char* schema_name = options->schema;
 
-	if (!schema_name || !*schema_name)
+	if (schema_name == NULL)
```

Comment:
In our case if we don't normalize one way or another we'd have to double check:

```
if (schema_name == NULL || schema_name[0] == 0)
```

Because for non present options the string pointers remain NULL: